File: server/internal/model/task.go
package model

import (
	"time"

	"do-it-server/pkg/utils"
	"gorm.io/gorm"
)

type TaskStatus int

const (
	StatusTodo       TaskStatus = 0
	StatusInProgress TaskStatus = 1
	StatusDone       TaskStatus = 2
)

type Task struct {
	ID          string     `gorm:"primaryKey;type:varchar(36)" json:"id"`
	Title       string     `gorm:"type:varchar(255);not null" json:"title"`
	Description string     `gorm:"type:text" json:"description"`
	Status      TaskStatus `gorm:"type:integer;default:0" json:"status"`
	Deadline    *time.Time `gorm:"type:datetime" json:"deadline,omitempty"`
	BeginTime   *time.Time `gorm:"type:datetime" json:"beginTime,omitempty"`
	EndTime     *time.Time `gorm:"type:datetime" json:"endTime,omitempty"`
	CreatedAt   time.Time  `gorm:"autoCreateTime" json:"createdAt"`
	UpdatedAt   time.Time  `gorm:"autoUpdateTime" json:"updatedAt"`
}

func (t *Task) BeforeCreate(tx *gorm.DB) error {
	if t.ID == "" {
		t.ID = utils.NewUUID()
	}
	return nil
}

File: server/src/models/task.ts
import { v4 as uuidv4 } from 'uuid';

export enum TaskStatus {
  TODO = 0,
  IN_PROGRESS = 1,
  DONE = 2,
}

export interface Task {
  id: string;
  title: string;
  description: string;
  status: TaskStatus;
  deadline: string | null;
  beginTime: string | null;
  endTime: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface TaskCreateInput {
  title: string;
  description?: string;
  deadline?: string;
  beginTime?: string;
  endTime?: string;
}

export interface TaskUpdateInput {
  title?: string;
  description?: string;
  status?: TaskStatus;
  deadline?: string | null;
  beginTime?: string | null;
  endTime?: string | null;
}
